tutorials/rag/agentic_rag.py

```diff
@@ -0,0 +1,274 @@
+"""
+This script demonstrates how to build a more agentic RAG pipeline using Humanloop and OpenAI's API.
+
+The model can choose to call the retrieval tool, or instead ask for clarification from the user.
+
+This leverages tool calling: https://platform.openai.com/docs/guides/function-calling
+
+It assumes that Prompts and Tools are managed in the client's code and logging and evals are done via Humanloop.
+"""
+
+import json
+from dotenv import load_dotenv
+import os
+from chromadb import chromadb
+from openai import OpenAI
+import pandas as pd
+from datetime import datetime
+from humanloop import Humanloop
+
+# Load .env file that contains API keys
+load_dotenv()
+
+# Init clients
+hl = Humanloop(api_key=os.getenv("HUMANLOOP_KEY"))
```

Comment:
are we standardizing on this client initialization? ferns docs have it as `humanloop = Humanloop()`

Reply:
We can do. I have a minor preference for `hl`